modules/src/erc721token.rs

```diff
@@ -0,0 +1,532 @@
+    }
+
+    pub fn transfer_from(&mut self, from: Address, to: Address, token_id: U256) {
+        self.core.transfer_from(from, to, token_id);
+    }
+
+    pub fn approve(&mut self, approved: Address, token_id: U256) {
+        self.core.approve(approved, token_id);
+    }
+
+    pub fn set_approval_for_all(&mut self, operator: Address, approved: bool) {
+        self.core.set_approval_for_all(operator, approved);
+    }
+
+    pub fn get_approved(&self, token_id: U256) -> Option<Address> {
+        self.core.get_approved(token_id)
+    }
+
+    pub fn is_approved_for_all(&self, owner: Address, operator: Address) -> bool {
+        self.core.is_approved_for_all(owner, operator)
+    }
+
+    pub fn mint(&mut self, to: Address, token_id: U256) {
+        if self.core.exists(&token_id) {
+            revert(Error::TokenAlreadyExists)
+        }
+
+        if to.is_zero() {
+            revert(ZeroAddress)
+        }
+
+        self.core.balances.add(&to, U256::from(1));
+        self.core.owners.set(&token_id, Some(to));
+    }
+
+    pub fn burn(&mut self, token_id: U256) {
+        self.core.assert_exists(&token_id);
+        if !self.core.is_approved_or_owner(caller(), token_id) {
+            revert(NotAnOwnerOrApproved);
+        }
+
+        let owner = self.core.owner_of(token_id);
+        self.core
+            .balances
+            .set(&owner, self.core.balance_of(owner) - U256::from(1));
+        self.core.owners.set(&token_id, None);
+        self.core.clear_approval(token_id);
+
+        Transfer {
+            from: Some(owner),
+            to: None,
+            token_id
+        }
+        .emit();
+    }
+}
+
+pub mod errors {
+    use odra::execution_error;
+
+    execution_error! {
+        pub enum Error {
+            TokenAlreadyExists => 35_000,
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::{Erc721TokenDeployer, Erc721TokenRef};
+    use crate::erc721::erc721::errors::Error;
+    use odra::test_env;
+    use odra::test_env::assert_exception;
+    use odra::types::{Address, U256};
+
+    const NAME: &str = "PlascoinNFT";
+    const SYMBOL: &str = "PLSNFT";
+    const BASE_URI: &str = "https://plascoin.org/";
+
+    struct TokenEnv {
+        token: Erc721TokenRef,
+        alice: Address,
+        bob: Address,
+        carol: Address,
+        zero: Address
+    }
+
+    fn setup() -> TokenEnv {
+        TokenEnv {
+            token: Erc721TokenDeployer::init(
+                NAME.to_string(),
+                SYMBOL.to_string(),
+                BASE_URI.to_string()
+            ),
+            alice: test_env::get_account(1),
+            bob: test_env::get_account(2),
+            carol: test_env::get_account(3),
+            zero: test_env::zero_address()
+        }
+    }
+
+    #[test]
+    fn mints_nft() {
+        // When deploy a contract with the initial supply.
+        let mut erc721_env = setup();
+
+        // And mint a token to Alice
```

Comment:
Tell Github Copilot to put dots at the end of every sentence.

Reply:
Fixed.